lib/evmone/baseline.cpp

```diff
@@ -0,0 +1,727 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2020 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "baseline.hpp"
+#include "execution_state.hpp"
+#include "instructions.hpp"
+#include <evmc/instructions.h>
+#include <memory>
+
+namespace evmone
+{
+namespace
+{
+using JumpdestMap = std::vector<bool>;
+
+JumpdestMap build_jumpdest_map(const uint8_t* code, size_t code_size)
+{
+    JumpdestMap m(code_size);
+    for (size_t i = 0; i < code_size; ++i)
+    {
+        const auto op = code[i];
+        if (op == OP_JUMPDEST)
+            m[i] = true;
+        else if (op >= OP_PUSH1 && op <= OP_PUSH32)
+            i += static_cast<size_t>(op - OP_PUSH1 + 1);
+    }
+    return m;
+}
+
+const uint8_t* op_jump(ExecutionState& state, const JumpdestMap& jumpdest_map) noexcept
+{
+    const auto dst = state.stack.pop();
+    if (dst >= jumpdest_map.size() || !jumpdest_map[static_cast<size_t>(dst)])
+    {
+        state.status = EVMC_BAD_JUMP_DESTINATION;
+        return &state.code[0] + state.code.size();
+    }
+
+    return &state.code[static_cast<size_t>(dst)];
+}
+
+template <size_t Len>
```

Comment:
For `dup` and `swap` the template parameter is the opcode (`load_push<1>` vs `dup<OP_DUP1>`), maybe worth to make it similar here for consistency?

Reply:
Good suggestion. Done separately as these instruction implementations are done outside of Baseline.